src/Symfony/Bundle/WebProfilerBundle/Tests/Controller/ProfilerControllerTest.php

```diff
@@ -366,6 +367,99 @@ public function provideCspVariants()
+    {
+        $this->assertDefaultPanel($expectedPanel, $profile);
+    }
+
+    public static function defaultPanelProvider(): \Generator
+    {
+        // Test default behavior
+        $profile = new Profile('xxxxxx');
+        $profile->addCollector($requestDataCollector = new RequestDataCollector());
+        yield [$requestDataCollector->getName(), $profile];
+
+        // Test exception
+        $profile = new Profile('xxxxxx');
+        $profile->addCollector($exceptionDataCollector = new ExceptionDataCollector());
+        $exceptionDataCollector->collect(new Request(), new Response(), new \DomainException());
+        yield [$exceptionDataCollector->getName(), $profile];
+    }
+
+    private function createController($profiler, $twig, $withCSP, array $templates = []): ProfilerController
+    {
+        $urlGenerator = $this->createMock(UrlGeneratorInterface::class);
+
+        if ($withCSP) {
+            $nonceGenerator = $this->createMock(NonceGenerator::class);
+            $nonceGenerator->method('generate')->willReturn('dummy_nonce');
+
+            return new ProfilerController($urlGenerator, $profiler, $twig, $templates, new ContentSecurityPolicyHandler($nonceGenerator));
+        }
+
+        return new ProfilerController($urlGenerator, $profiler, $twig, $templates);
+    }
+
+    public function testDumpPanelExceptionPriority()
+    {
+        $exceptionDataCollector = new ExceptionDataCollector();
+        $exceptionDataCollector->collect(new Request(), new Response(), new \DomainException());
+
+        $dumpDataCollector = $this->createDumpDataCollector();
+
+        $profile = new Profile('xxxxxx');
+        $profile->setCollectors([$exceptionDataCollector, $dumpDataCollector]);
+
+        $this->assertDefaultPanel($exceptionDataCollector->getName(), $profile);
+    }
+
+    public function testDumpPanelWhenDefinedAfterwards()
+    {
+        $exceptionDataCollector = new ExceptionDataCollector();
+        $exceptionDataCollector->collect(new Request(), new Response(), new \DomainException());
+
+        $dumpDataCollector = $this->createDumpDataCollector();
+        $dumpDataCollector
+            ->expects($this->atLeastOnce())
+            ->method('getDumpsCount')
+            ->willReturn(1)
+        ;
+
+        $profile = new Profile('xxxxxx');
+        $profile->setCollectors([$dumpDataCollector, $exceptionDataCollector]);
+
+        $this->assertDefaultPanel($exceptionDataCollector->getName(), $profile);
+    }
+
+    public function testDumpPanel()
+    {
+        $dumpDataCollector = $this->createDumpDataCollector();
+        $dumpDataCollector
+            ->expects($this->atLeastOnce())
+            ->method('getDumpsCount')
+            ->willReturn(1)
+        ;
+
+        $profile = new Profile('xxxxxx');
+        $profile->addCollector($dumpDataCollector);
+
+        $this->assertDefaultPanel($dumpDataCollector->getName(), $profile);
+    }
+
+    /**
+     * @return MockObject<DumpDataCollector>
```

Comment:
The right type here is `MockObject&DumpDataCollector` actually. PHPUnit does not make the MockObject interface generic.